config/covid-19/nools-extras.js

```diff
@@ -0,0 +1,47 @@
+
+function getField(report, fieldPath) {
+  const parts = (fieldPath || '').split('.');
+
+  return ['fields', ...parts]
+    .reduce((prev, fieldName) => {
+      if (prev === undefined) {
+        return undefined;
+      }
+      return prev[fieldName];
+    }, report);
+}
+
+function isFormArraySubmittedInWindow(reports, formArray, start, end, count) {
+  const reportsFound = (reports || []).filter(report => {
+    return (formArray || []).includes(report.form)
+      && report.reported_date >= start
+      && report.reported_date <= end;
+  });
+
+  if (count) {
+    return reportsFound.length >= count;
+  }
+
+  return !!reportsFound.length;
+}
+
+function getTimeForMidnight(originalDate) {
+  const date = new Date(originalDate);
+  date.setHours(0);
+  date.setMinutes(0);
+  date.setSeconds(0);
+  date.setMilliseconds(0);
+  return date;
+}
+
+function addDays(date, days) {
```

Comment:
A bummer we don't integrate these into cht-core or a common medic-module so each config doesn't have to reimplement them and such retest them.

Reply:
I wonder if it's the same code across production instances, [we recently added an api](https://docs.communityhealthtoolkit.org/apps/reference/_partial_cht_api/) for contact summary, targets and tasks where we can put common functions as app services requires them, something to explore with them first and see what other things to include.